src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -159,8 +159,20 @@ public void visitToken(DetailAST ast) {
      * @return true if a method has a javadoc comment.
      */
     private static boolean hasJavadocComment(DetailAST methodDef) {
-        final DetailAST modifiers = methodDef.findFirstToken(TokenTypes.MODIFIERS);
-        return modifiers.branchContains(TokenTypes.BLOCK_COMMENT_BEGIN);
+        return hasJavadocCommentOnToken(methodDef, TokenTypes.MODIFIERS)
+                || hasJavadocCommentOnToken(methodDef, TokenTypes.TYPE);
+    }
```

Comment:
hmm, we already have https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java#L194 that cover all cases where user is using comments but not javadoc. We need to recheck on found comment that it is actually a javadoc.
Attention: block comment is not always a javadoc comment.

Reply:
@romani
There is an issue for this:
https://github.com/checkstyle/checkstyle/issues/5450

Reply:
@romani We moved it to another issue to keep this PR simple.

Reply:
this method doesn't work with javadoc on annotated method... i made some tests on this